src/impl/featuresRegistry.ts

```diff
@@ -0,0 +1,30 @@
+import { Feature } from "../interfaces/feature";
+
+export class FeaturesRegistry {
+
+    private _storage = new Map<string[], Feature>();
+
+    constructor(features: Feature[]) {
+        features?.forEach(f => this.put(f));
+    }
+
+    public put(feature: Feature) {
+        this._storage.set(feature.regKey, feature);
+    }
+
+    public get(...name: string[]) {
```

Comment:
Why not use a usual array as an argument?
`public get(key: string[]){..}`
maybe
`type regKeyType = string[];`
`public get(key: regKeyType){..}`